core/generated-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/json.kt

```diff
@@ -859,42 +867,103 @@ private val valueTypes =
 
 internal fun KlaxonJson.encodeRow(frame: ColumnsContainer<*>, index: Int): JsonObject? {
     val values = frame.columns().map { col ->
-        when {
-            col is ColumnGroup<*> -> encodeRow(col, index)
-            col is FrameColumn<*> -> encodeFrame(col[index])
-            col.isList() -> {
-                col[index]?.let { array(it as List<*>) } ?: array()
-            }
+        when (col) {
+            is ColumnGroup<*> -> encodeRow(col, index)
+            is FrameColumn<*> -> encodeFrame(col[index])
+            else -> encodePrimitiveData(col, index)
```

Comment:
let's call it `encodeValue`?  Name primitive made me think that some cases were removed (List, else)